server/src/main/java/org/elasticsearch/index/codec/vectors/cluster/KMeansLocal.java

```diff
@@ -170,16 +179,32 @@ private void computeNeighborhoods(float[][] centers, List<int[]> neighborhoods,
             }
         }
 
+        float[] scores = new float[clustersPerNeighborhood];
         for (int i = 0; i < k; i++) {
             NeighborQueue queue = neighborQueues.get(i);
             int neighborCount = queue.size();
             int[] neighbors = new int[neighborCount];
-            queue.consumeNodes(neighbors);
-            neighborhoods.set(i, neighbors);
+            float maxIntraDistance = queue.consumeNodesWithWorstScore(neighbors, scores);
+            // Sort neighbors by their score
+            for (int j = 0; j < neighborCount; j++) {
```

Comment:
Why not populate the array sing the pop method of the priority queue? 

```
            for (int j = neighborCount - 1; j >= 0; j--) {
                neighbors[j] = queue.pop();
            }
```

Reply:
@iverase let me benchmark this